daps-xslt/asciidoc/postprocess.xsl

```diff
@@ -170,4 +172,89 @@
     </xsl:if>
   </xsl:template>
 
+
+  <!--
+  A correctly-formatted AsciiDoc example with a screen would look like this
+  (be aware the − characters are minuses, not normal hyphens, -, do not copy):
+
+      .Title of example 1 (correct)
+      ====
+      −−−−
+      command -option='value'
+      −−−−
+      ====
+
+  => <example><title>Title...</title><screen>command...</screen></example>
+
+
+
+  However, authors are creative and came up with these variants as well:
+
+      .Title of example 2 (bad)
+      −−−−
+      command -option='value'
+      −−−−
+
+  => <formalpara><title>Title...</title><screen>command...</screen></formalpara>
+
+
+      ====
+      .Title of example 3 (bad)
+      −−−−
+      command -option='value'
+      −−−−
+      ====
+
+  => <informalexample><formalpara><title>Title...</title><screen>command...</screen></formalpara></informalexample>
+
+
+  While the syntax of example (2) does not support more elements before/after
+  the screen, example (3) unfortunately does. For example, there could be
+  multiple formalparas that are generated within the `====` construct of
+  example (3), which might then have to be handled by both the rule for
+  example (3) and the rule for example (2).
+
+  -->
+
+  <xsl:template match="d:formalpara[d:para/d:screen|
+                                    d:para/d:literallayout[@class='monospaced']]">
+    <xsl:element name="example" namespace="&db5ns;">
+      <xsl:apply-templates select="d:title"/>
+      <!-- The output of the following two apply-templates is unlikely to be
+      in the correct order but at least we get it all into the output
+      document. However, it is rather unlikely that there is any additional
+      output in any case. -->
+      <xsl:apply-templates select="d:para" mode="exclude-verbatim"/>
+      <xsl:apply-templates select="d:para/d:screen|
+                                   d:para/d:literallayout[@class='monospaced']"/>
+    </xsl:element>
+  </xsl:template>
+
+  <xsl:template match="d:informalexample[d:*[1][self::d:formalpara]
+                                         [d:para/d:screen|
+                                          d:para/d:literallayout[@class='monospaced']]]">
+    <xsl:element name="example" namespace="&db5ns;">
+      <xsl:apply-templates select="@*"/>
+      <xsl:apply-templates select="d:formalpara[1]/d:title"/>
+      <!-- The output of the following two apply-templates is unlikely to be
+      in the correct order but at least we get it all into the output
+      document. However, it is rather unlikely that there is any additional
+      output in any case. -->
+      <xsl:apply-templates select="d:formalpara[1]/d:para" mode="exclude-verbatim"/>
```

Comment:
You're right, the line is not necessary with our current examples.

However, it was meant as a defensive measure, in case there are other similar constructs that we have not yet thought about.

Reply:
Okay, I see. Maybe add that aspect in the comment?